rpc/transaction_test.go

```diff
@@ -170,41 +190,51 @@ func TestTransactionReceipt(t *testing.T) {
+		BlockHash:       utils.TestHexToFelt(t, "0x4ae5d52c75e4dea5694f456069f830cfbc7bec70427eee170c3385f751b8564"),
+		BlockNumber:     52767,
+		ActualFee: FeePayment{
+			Amount: utils.TestHexToFelt(t, "0x16409a78a10b00"),
+			Unit:   UnitStrk,
+		},
 		Type:            "INVOKE",
 		ExecutionStatus: TxnExecutionStatusSUCCEEDED,
 		FinalityStatus:  TxnFinalityStatusAcceptedOnL1,
 		MessagesSent:    []MsgToL1{},
 		Events: []Event{
 			{
-				FromAddress: utils.TestHexToFelt(t, "0x37de00fb1416936b3074fc78bcc811d83046009b162c4a822ce84dabedd0ea9"),
+				FromAddress: utils.TestHexToFelt(t, "0x243d436e1f7cea085aaa42834975488029b1ebf67cea1d2e86f7de58e7d34a3"),
 				Data: []*felt.Felt{
-					utils.TestHexToFelt(t, "0x0"),
-					utils.TestHexToFelt(t, "0x35b32bb4a1969175fb14b6c09838d1b3200724cc4d2b0891be319764021f5ac"),
-					utils.TestHexToFelt(t, "0xe9"),
-					utils.TestHexToFelt(t, "0x0"),
+					utils.TestHexToFelt(t, "0x3028044a4c4df95c0b0a907307c6feffa76b9c38e83088ade29b186a250eb13"),
+					utils.TestHexToFelt(t, "0x3"),
+					utils.TestHexToFelt(t, "0x17a393a5e943cec833c8a8f4cbbf7c58361fb2fdd9caa0c36d901eedec4938e"),
+					utils.TestHexToFelt(t, "0x776731d30bd922ac0390edfc664ed31b232aa7c7ce389c333e34c6b32957532"),
+					utils.TestHexToFelt(t, "0x40851db0ebaebb9f8a18eda25005c050793f2a69e9e7d1f44bc133752898918"),
+				},
+				Keys: []*felt.Felt{
+					utils.TestHexToFelt(t, "0x243d436e1f7cea085aaa42834975488029b1ebf67cea1d2e86f7de58e7d34a3"),
 				},
-				Keys: []*felt.Felt{utils.TestHexToFelt(t, "0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9")},
 			},
 			{
-				FromAddress: utils.TestHexToFelt(t, "0x33830ce413e4c096eef81b5e6ffa9b9f5d963f57b8cd63c9ae4c839c383c1a6"),
+				FromAddress: utils.TestHexToFelt(t, "0x243d436e1f7cea085aaa42834975488029b1ebf67cea1d2e86f7de58e7d34a3"),
 				Data: []*felt.Felt{
-					utils.TestHexToFelt(t, "0x61c6e7484657e5dc8b21677ffa33e4406c0600bba06d12cf1048fdaa55bdbc3"),
-					utils.TestHexToFelt(t, "0x2e28403d7ee5e337b7d456327433f003aa875c29631906908900058c83d8cb6"),
+					utils.TestHexToFelt(t, "0x6016d919abf2ddefe03dacc2ff5c8f42eb80cf65add1e90dd73c5c5e06ef3e2"),
+					utils.TestHexToFelt(t, "0x1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8"),
+					utils.TestHexToFelt(t, "0x16409a78a10b00"),
+					utils.TestHexToFelt(t, "0x0"),
+				},
+				Keys: []*felt.Felt{
+					utils.TestHexToFelt(t, "0x15bd0500dc9d7e69ab9577f73a8d753e8761bed10f25ba0f124254dc4edb8b4"),
 				},
-				Keys: []*felt.Felt{utils.TestHexToFelt(t, "0xf806f71b19e4744968b37e3fb288e61309ab33a782ea9d11e18f67a1fbb110")},
 			},
 		},
 		ExecutionResources: ExecutionResources{
 			ComputationResources: ComputationResources{
-				Steps:          217182,
-				MemoryHoles:    6644,
-				PedersenApps:   2142,
-				RangeCheckApps: 8867,
-				BitwiseApps:    900,
-				ECDSAApps:      1,
+				Steps:          5774,
+				PedersenApps:   24,
+				ECOPApps:       3,
+				RangeCheckApps: 152,
```

Comment:
We should also update the test to require the entire receipt to be equal to what we get from the full node:
require.Equal(t, txReceiptWithBlockInfo, test.ExpectedResp)

Reply:
I'm getting an error here, where the `TransactionReceipt` object is nil. And I'm not able to figure out how to fix this.
I checked for both mock and testnet, 

Getting: 
```json
actual  : *rpc.TransactionReceiptWithBlockInfo(&rpc.TransactionReceiptWithBlockInfo{TransactionReceipt:
rpc.TransactionReceipt(nil), 
BlockHash:(*felt.Felt)(0x140004e6b40), 
BlockNumber:0xce1f})
```

You see, the transactionReceipt is nil. 
There seems to be an issue while marshalling the data, between the `CommonTransactionReceipt` struct and `TransactionReceipt` interface. But I'm not able to find the exact point where it's going wrong.

Expected :-
```json
expected: rpc.TransactionReceiptWithBlockInfo(rpc.TransactionReceiptWithBlockInfo{TransactionReceipt:rpc.InvokeTransactionReceipt{TransactionHash:(*felt.Felt)(0x1400050fec0), ActualFee:rpc.FeePayment{Amount:(*felt.Felt)(0x1400050fee0), Unit:"FRI"}, ExecutionStatus:"SUCCEEDED", FinalityStatus:"ACCEPTED_ON_L1", Type:"INVOKE", MessagesSent:[]rpc.MsgToL1{}, RevertReason:"", Events:[]rpc.Event{rpc.Event{FromAddress:(*felt.Felt)(0x1400050ff00), Keys:[]*felt.Felt{(*felt.Felt)(0x1400050ffc0)}, Data:[]*felt.Felt{(*felt.Felt)(0x1400050ff20), (*felt.Felt)(0x1400050ff40), (*felt.Felt)(0x1400050ff60), (*felt.Felt)(0x1400050ff80), (*felt.Felt)(0x1400050ffa0)}}, rpc.Event{FromAddress:(*felt.Felt)(0x1400050ffe0), Keys:[]*felt.Felt{(*felt.Felt)(0x140004940a0)}, Data:[]*felt.Felt{(*felt.Felt)(0x14000494020), (*felt.Felt)(0x14000494040), (*felt.Felt)(0x14000494060), (*felt.Felt)(0x14000494080)}}}, ExecutionResources:rpc.ExecutionResources{ComputationResources:rpc.ComputationResources{Steps:5774, MemoryHoles:0, RangeCheckApps:152, PedersenApps:24, PoseidonApps:0, ECOPApps:3, ECDSAApps:0, BitwiseApps:0, KeccakApps:0, SegmentArenaBuiltin:0}, DataAvailability:rpc.DataAvailability{L1Gas:0x0, L1DataGas:0x80}}}, BlockHash:(*felt.Felt)(0x14000494300), BlockNumber:0xce1f})
```